packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol

```diff
@@ -696,4 +708,220 @@ contract BridgeFacet is BaseConnextFacet {
       }
     }
   }
+
+  /**
+   * @notice Uses Aave Portals to provide fast liquidity
+   */
+  function _executePortalTransfer(
+    bytes32 _transferId,
+    uint256 _fastTransferAmount,
+    address _local,
+    address _router
+  ) internal returns (uint256, address) {
+    // Calculate local to adopted swap output if needed
+    (uint256 userAmount, address adopted) = AssetLogic.calculateSwapFromLocalAssetIfNeeded(_local, _fastTransferAmount);
+
+    IAavePool(s.aavePool).mintUnbacked(adopted, userAmount, address(this), AAVE_REFERRAL_CODE);
+
+    // Improvement: Instead of withdrawing to address(this), withdraw directly to the user or executor to save 1 transfer
+    IAavePool(s.aavePool).withdraw(adopted, userAmount, address(this));
+
+    s.aavePortalsTransfers[_transferId] = userAmount;
+
+    emit AavePortalMintUnbacked(_transferId, _router, adopted, userAmount);
+
+    return (userAmount, adopted);
+  }
+
+  /**
+   * @notice Parses the message and process the transfer
+   * @dev Will mint the tokens if the token originates on a remote origin
+   * @return The message amount
+   * @return The message token
+   * @return The message transfer id
+   */
+  function _reconcileProcessMessage(bytes memory _message)
```

Comment:
Personally i'm not a fan of these "one-time-use" extension functions that we're using to avoid stack too deep errors. Would strongly prefer to find a way to scope this within the `_reconcile` method itself.

8 variables are declared here, and 7 are declared in the parent function including args - which keeps us just under the ref stack size limit of 16/

If we want (or need) to reduce it further, you could easily just scope this method's logic in a block, and keep only the 4 'return' variables `(uint256 amount, address local, bytes32 transferId, bytes32 canonical)` outside of the scope. At 11 vars, should be good.

cc @LayneHaber in case you have any thoughts on this. This is certainly IMHO, but i'm pushing to avoid this 'extension' pattern where possible

Reply:
Just noticed we have this `_reconcileProcessPortal` method as well, which has 6 variables. That should also fit within the `_reconcile` method as long as we scope that method's logic entirely.

Reply:
oh wait there's also `_calculatePortalRepayment` LOL

ok i guess this is nbd for now. still think you could probs fit all of it in 1 method but nbd

Reply:
i also think this could go into a separate PR of just code-styling and best practices clean up